pydantic/exceptions.py

```diff
@@ -103,3 +75,36 @@ class Missing(ValueError):
 
 class Extra(ValueError):
     pass
+
+
+def display_errors(errors):
+    display = []
+
+    for error in errors:
+        display.extend([
+            ' -> '.join(error['loc']),
+            f'  {error["msg"]} (type={error["type"]})',
+        ])
+
+    return '\n'.join(display)
+
+
+def flatten_errors(errors, *, loc=None):
```

Comment:
Can you please describe why you want to see this as generator?

Reply:
it will be much cleaner and more explicit with less code.

```py
def flatten_errors(errors, *, loc=None):
    for error in errors:
        if isinstance(error, Error):
            if isinstance(error.exc_info, ValidationError):
                yield from flatten_errors(error.exc_info.errors, loc=error.loc)
            else:
                yield {
                    'loc': error.loc if loc is None else loc + error.loc,
                    'msg': error.msg,
                    'type': error.type_,
                }
        elif isinstance(error, list):
            yield from flatten_errors(error)
        else:
            raise RuntimeError(f'Unknown error object: {error}')
```

Reply:
great! I forgot about `yield from`